zerver/lib/db.py

```diff
@@ -21,6 +21,7 @@ def wrapper_execute(self: CursorObj,
         duration = stop - start
         self.connection.queries.append({
             'time': "%.3f" % (duration,),
+            'sql': sql,
```

Comment:
I'm a bit reluctant to make this change unconditionally, though I'm not sure if it's for good reason.  I think this can get expensive in certain leak situations or if the `sql` is very long for a database query we do in a loop (our query to fetch a single user can be like 1KB long with all the fields to fetch).  So 1K of those could end up making a 1MB object here after 1K queries, which seems not great. 

Some background is that Django's debug cursor (which one can enable) stores every query this way  (https://docs.djangoproject.com/en/3.0/faq/models/#how-can-i-see-the-raw-sql-queries-django-is-running) and I seem to recall reading they didn't track the data unconditionally for performance reasons.

I think it's probably worth taking the opportunity to make our `connection.queries` logging structure a little smarter:
* We don't need to record the time for every query; we can just have a `total_queries` and a `total_time` and a list of just the "interesting queries".
* We can just keep the SQL for all queries that take more than some level like 25ms (I'd make it a setting, maybe `SQL_PERF_LOGGING_MIN_TIME` so we can tweak it easily) and plan to use that list (rather than the top-3 model).  

With those changes, I think it'd be safe performance-wise to have this bit of code running unconditionally.